### PR TITLE
Enhancement for MeshSync Settings Page

### PR DESCRIPTION
**Pull request by dhruv0000**
Signed-off-by: dhruv0000 <patel.4@iitj.ac.in>
...and potentially Config Wizard.

**Description**

This PR:
1. Gives user the ability to do few step individually
     a. Deploy MeshSync
     b. connect to broker

2. Adds one more state `CONNECTED` to show if Meshery Server is CONNECTED  to NATS.

https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=4%3A5825

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
A few merge conflicts have emerged. Let's use the NATS logo to represent NATS.

**Comment by sudo-NithishKarthik** [comment]
> Let's use the NATS logo to represent NATS.

@leecalcote , i don't think their logo is something that will fit in the chip that we are using 🤔 

Screenshot 2021-08-21 at 12 45 12 PM

**Comment by leecalcote** [comment]
@sudo-NithishKarthik here you go - https://raw.githubusercontent.com/cncf/artwork/master/projects/nats/icon/color/nats-icon-color.svg

**Review by leecalcote — DISMISSED** [review]
Please use the provided NATS logo SVG

**Comment by sudo-NithishKarthik** [comment]
There are some changes yet to be done in UI, i will make them today, and we should be good to go.

File: ui/components/MeshConfigComponent.js [hunk 1]
```diff
@@ -451,30 +484,71 @@ class MeshConfigComponent extends React.Component {
...
+
+  handleMeshSyncClick = () => {
+    this.props.updateProgress({ showProgress : true });
+    const self = this;
+
+    deployMeshSync().subscribe({
+      next: (res) => {
+        if(res.deployMeshsync === "PROCESSING") {
+          this.props.updateProgress({ showProgress: false });
+          this.props.enqueueSnackbar(`Meshsync deployment in progress`, {
```
Comment (leecalcote) [inline ui/components/MeshConfigComponent.js:533]:
```suggestion
          this.props.enqueueSnackbar(`MeshSync deployment in progress`, {
```

**Review by leecalcote — COMMENTED** [review]
Does this also address the slowness of this subscription to start ( 4 or 5 seconds for actual Meshery Operator status to show) or will that be separately handled?

**Comment by leecalcote** [comment]
Merge conflicts have come about.

**Comment by sudo-NithishKarthik** [comment]
> Does this also address the slowness of this subscription to start ( 4 or 5 seconds for actual Meshery Operator status to show) or will that be separately handled?

No, I think it's better to handle it separately.

**Comment by sudo-NithishKarthik** [comment]
Screenshot 2021-08-31 at 5 39 13 PM

**Comment by leecalcote** [comment]
a couple conflicts have arisen

**Comment by dhruv0000** [comment]
Trying out broadcasting.

**Comment by dhruv0000** [comment]
Dependency: https://github.com/layer5io/meshkit/pull/103

**Comment by dhruv0000** [comment]
wow. Broadcasting was the thing. Subs are working like a charm.

**Comment by sudo-NithishKarthik** [comment]
Awesome  @dhruv0000  🙌

File: ui/components/MeshConfigComponent.js [hunk 2]
```diff
@@ -662,26 +735,46 @@ class MeshConfigComponent extends React.Component {
...
+                <List>
+                  <ListItem>
+                    <Tooltip
+                      title={NATSState === "CONNECTED" ? `Reconnect NATS` : "Not Available"}
+                      aria-label="nats"
+                    >
+                      <Chip
+                        label={"NATS"}
+                        onClick={self.handleNATSClick}
+                        icon={<img src="https://raw.githubusercontent.com/cncf/artwork/master/projects/nats/icon/color/nats-icon-color.svg" className={classes.icon} />}
```
Comment (leecalcote) [inline ui/components/MeshConfigComponent.js:768]: All assets need to be locally stored.

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
Depend PR https://github.com/layer5io/meshkit/pull/103 must be merged first, yes? MeshKit released and dependency updated here?

File: cmd/main.go [hunk 3]
```diff
@@ -240,11 +241,15 @@ func main() {
 
 	h := handlers.NewHandlerInstance(hc, meshsyncCh, log, brokerConn)
 
+	b := broadcast.NewBroadcaster(100)
```
Comment (leecalcote) [inline cmd/main.go:244]: 100 is the channel buffer length, correct? What's the significance of 100? What factors lead us to this size?

Comment (dhruv0000) [reply cmd/main.go:244]: Nothing particular significance.

**Comment by dhruv0000** [comment]
> Dependent PR [layer5io/meshkit#103](https://github.com/layer5io/meshkit/pull/103) must be merged first, yes? MeshKit released and dependency updated here?

`go.mod` is pointing to my meshkit repo (temporarily) , so this PR is good to merge. If the MeshKit PR is merged first, can update the `go.mod` here.

File: ui/components/MeshConfigComponent.js (hunk 2 above)
Comment (dhruv0000) [reply ui/components/MeshConfigComponent.js]: Added icon locally.

**Comment by leecalcote** [comment]
> > Dependent PR [layer5io/meshkit#103](https://github.com/layer5io/meshkit/pull/103) must be merged first, yes? MeshKit released and dependency updated here?
> 
> 
> 
> `go.mod` is pointing to my meshkit repo (temporarily) , so this PR is good to merge. If the MeshKit PR is merged first, can update the `go.mod` here.

Thank you. We always want to avoid pointing to third-party repos when possible. Let's land the MeshKit PR and release.

File: ui/components/MeshConfigComponent.js (hunk 2 above)
Comment (leecalcote) [reply ui/components/MeshConfigComponent.js]: Perfect. Thank you.

**Review by sudo-NithishKarthik — APPROVED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
CITY

File: ui/components/MeshConfigComponent.js [hunk 4]
```diff
@@ -147,10 +149,16 @@ class MeshConfigComponent extends React.Component {
       operatorVersion : "N/A",
       meshSyncInstalled : false,
       meshSyncVersion : "N/A",
-      NATSInstalled : false,
+      NATSState : "UNKNOWN",
```
Comment (leecalcote) [inline ui/components/MeshConfigComponent.js:152]: Gents, let's avoid all caps in the UI, unless the text is a label.

Comment (sudo-NithishKarthik) [reply ui/components/MeshConfigComponent.js:152]: Ok

**Pull request merged**